docs/src/man/comparisons.md

```diff
@@ -1,77 +1,204 @@
+two lines of code are functionally equivalent:
+
+```julia
+df[findfirst(==('c'), df.id), :]
+df[df.id .== 'c', :]
+```
+
+### Common operations
+
+| Operation                | pandas                                                | DataFrames.jl                           |
+| :----------------------- | :---------------------------------------------------- | :-------------------------------------- |
+| Reduce multiple values   | `df['z'].mean(skipna = False)`                        | `mean(df.z)`                            |
+|                          | `df['z'].mean()`                                      | `mean(skipmissing(df.z))`               |
+|                          | `df[['z']].agg(['mean'])`                             | `combine(df, :z => mean ∘ skipmissing)` |
+| Add new columns          | `df.assign(x_mean = df['x'].mean())`                  | `transform(df, :x => mean => :x_mean)`  |
+| Rename columns           | `df.rename(columns = {'x': 'x_new'})`                 | `rename(df, :x => :x_new)`              |
+| Pick & transform columns | `df.assign(x_mean = df['x'].mean())[['x_mean', 'y']]` | `select(df, :x => mean, :y)`            |
+| Sort rows                | `df.sort_values(by = ['x'])`                          | `sort(df, :x)`                          |
+
+Note that Julia propagates `missing` data by default for safety reasons. The `skipmissing` function
+can be used to remove missing data. See more details at the [Additional Differences](@ref) section below.
+
+### Grouping data and aggregation
+
+| Operation                    | pandas                                                                                    | DataFrames.jl                               |
+| :--------------------------- | :---------------------------------------------------------------------------------------- | :------------------------------------------ |
+| Aggregate by groups          | `df.groupby('grp')['x'].mean().reset_index()`                                             | `combine(groupby(df, :grp), :x => mean)`    |
+| Aggregate and add columns    | `df.join(df.groupby('grp')['x'].mean(), on='grp', rsuffix='_mean')`                       | `transform(groupby(df, :grp), :x => mean)`  |
+| Aggregate and select columns | `df.join(df.groupby('grp')['x'].mean(), on='grp', rsuffix='_mean')[['grp','x_mean','y']]` | `select(groupby(df, :grp), :x => mean, :y)` |
+
+### More advanced commands
+
+| Operation                 | pandas                                                                       | DataFrames.jl                                                 |
+| :------------------------ | :--------------------------------------------------------------------------- | :------------------------------------------------------------ |
+| Complex Function          | `df[['z']].agg(lambda v: np.mean(np.cos(v)))`                                | `combine(df, :z => v -> mean(cos, skipmissing(v)))`           |
```

Comment:
The pandas code here drops the data frame, so in data frames it would be `mean(cos.(skipmissing(df.z)))` - but then I am not sure we need to show it as we have a similar example above. If we want to keep showing it it would be good to give an example that retains the object being a `DataFrames` there

Reply:
Well, to calculate a single value then these are equivalent:

```
>>> np.mean(np.cos(df.z))
0.07081972179856785
```

```
julia> mean(cos.(skipmissing(df.z)))
0.07081972179856785
```

But then it's not a "Complex Function" anymore. If I understand the purpose of this operation properly, we want illustrate how to apply a custom function (other than the regular `min`, `mean`, etc.) when aggregating a data frame.

As related to my other note, do we want to make the result match exactly?  I would rather not. In pandas, I can convert a `Series` to a `DataFrame` and then transpose it to come up with the same shape as in DataFrames.jl. And then if the column name needs to be adjusted then you have a 100-char code that is so obfuscated that people wouldn't even understand.

Having said these, I think it's better to keep it as-is. Let me know if you agree, and if not, would you mind suggesting another viable solution?